hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -278,6 +282,20 @@ private Object proxyOrLoad(
 				return managed;
 			}
 
+			//Check for presence of the required data in 2LC, as in this case there is no compelling reason to
+			//instantiate a lazily initialized enhanced proxy: might as well return it initialized.
```

Comment:
The concern here is the overhead of the L2C interactions when we do not necessarily have to.  Even the look into the cache to find an existing entry can be prohibitive depending on the underlying cache.  

That is perhaps not compelling in your use case, but this is something specifically discussed before.

Reply:
How about a new method `Object getForProxy(SharedSessionContractImplementor session, Object key, long txTimestamp) throws CacheException` in the cache SPI `EntityRegionAccessStrategy` to allow handling this case?
The cache implementor can then decide to return null if the access would be too expensive.

Reply:
@sebersole I agree but how compelling is this concern in this context? This extra 2LC cost is only applied for entities (and eventsources) which actually do have caching enabled, so isn't this already something that the user has been explicitly opting-in?

How would you do it otherwise, do you think the "on proxy initialization hit 2LC" approach would fare better? Or are you thinking of having some configuration?

@beikov If we are to explore that direction, it would need to be more complicted than "just returning null" as we need to know if the lookup was actually attempted - so to ensure that eventually when we're sure the data is needed we know if we should _really_ attempt the 2LC load, or if we rather shouldn't bother as this was already happened.

As you can see in my PR, I make sure that the 2LC lookup happens _at most once_.